apps/src/templates/teacherDashboard/AddSectionDialog.jsx

```diff
@@ -0,0 +1,159 @@
+import React, {Component, PropTypes} from 'react';
+import { connect } from 'react-redux';
+import $ from 'jquery';
+import BaseDialog from '../BaseDialog';
+import AddInitialStudentsView from './AddInitialStudentsView';
+import EditSectionForm from "./EditSectionForm";
+
+import i18n from '@cdo/locale';
+import { updateSection } from './teacherSectionsRedux';
+const initialState = {
+  loginType: undefined,
+  name: '',
+  grade: '',
+  extras: 'yes',
+  pairing: 'yes',
+};
+
+export class AddSectionDialog extends Component {
+  static propTypes = {
+    handleClose: PropTypes.func,
+    isOpen: PropTypes.bool,
+    // Provided by Redux
+    updateSection: PropTypes.func.isRequired,
+  };
+
+  state = {
+    ...initialState,
+  };
+
+  handleClose = () => {
+    this.props.handleClose();
+    this.setState(initialState);
+  };
+
+  handleLoginChoice = (loginType) => {
+    this.setState({loginType});
+  };
+
+  handleNameChange = (name) => {
+    this.setState({name});
+  };
+
+  handleGradeChange = (grade) => {
+    this.setState({grade});
+  };
+
+  handleExtrasChange = (extras) => {
+    this.setState({extras});
+  };
+
+  handlePairingChange = (pairing) => {
+    this.setState({pairing});
+  }
+
+  onClickEditSave = () => {
+    const {updateSection} = this.props;
+    //const persistedSection = false;
+
+    const selectedAssignment = this.assignment.getSelectedAssignment();
+    const data = {
+      //id: null,
+      name: this.state.name,
+      login_type: this.state.loginType,
+      grade: this.state.grade,
+      stage_extras: this.state.extras === 'yes' ? true : false,
+      pairing_allowed: this.state.pairing === 'yes' ? true : false,
+      course_id: selectedAssignment ? selectedAssignment.courseId : null,
+    };
+
+    if (selectedAssignment && selectedAssignment.scriptId) {
+      data.script = {
+        id: selectedAssignment.scriptId
+      };
+    }
+
+    const suffix = '';
+    const sectionId = -1; // When it's a new section
+
+    $.ajax({
+      url: `/v2/sections${suffix}`,
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify(data),
+    }).done(result => {
+      updateSection(sectionId, result);
+      // close modal after save
+      this.handleClose();
+    }).fail((jqXhr, status) => {
+      // We may want to handle this more cleanly in the future, but for now this
+      // matches the experience we got in angular
+      alert(i18n.unexpectedError());
+      console.error(status);
+    });
+  };
+
+  renderContent() {
+    const title = i18n.newSection();
+    if (!this.state.loginType) {
+      return (
+        <AddInitialStudentsView
+          title={title}
+          handleLoginChoice={this.handleLoginChoice}
+          handleCancel={this.handleClose}
+        />
+      );
+    } else {
+      return (
+        <EditSectionForm
+          title={title}
+          assignmentRef = {(element) => this.assignment = element}
```

Comment:
nit: you have spaces around the `=` that you dont have elsewhere

Also, I would call this `ref` rather than assignmentRef. Ref is a specific concept in React (https://facebook.github.io/react/docs/refs-and-the-dom.html). Though I think it makes sense to me that this works even with the name `assignmentRef`, you might be missing out on some things doing it this way, and also convention is `ref`